### PR TITLE
Add an API function to get all item types (as opposed to a page by page basis).

### PR DESCRIPTION
**Pull request by officialcwang**
This is easier than handling the requests on the front-end side. 
All tests pass.

### CODE REVIEW

File: capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java [hunk 1]
```diff
@@ -124,6 +124,21 @@ public static void createUser(long userId, String userName, String email)
     dbClient.write(Arrays.asList(mutation));
   }
 
+  public static List<String> getItemTypes() throws SpannerException {
```
Comment (chen-dawn) [inline capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java:127]: where are you using this function?

File: capstone/client/src/components/ListPage.js [hunk 2]
```diff
@@ -184,12 +184,12 @@ class ListPage extends Component {
...
         errorMessage: "There was an error saving your list.",
       })
-    });   
+    });
   }
 
   getItemTypes = () => {
     axios.get(
-      '/api/v1/get-item-types?page=0'
+      '/api/v1/get-item-types?page=0'
```
Comment (chen-dawn) [inline capstone/client/src/components/ListPage.js:192]: Better to use params here: 
https://kapeli.com/cheat_sheets/Axios.docset/Contents/Resources/Documents/index

File: capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java (hunk 1 above)
Comment (officialcwang) [reply capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java:127]: Using this function in ListPage.js. Looping through the GET requests on the front-end turns out to be a lot more of a hassle.

Comment (officialcwang) [reply capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java:127]: If this goes through, I can just use axios.get('/api/v1/get-item-types') in ListPage.js

Comment (chen-dawn) [reply capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java:127]: Why is it a hassle? You can just change this line here to use your new function right? 
https://github.com/googleinterns/step39-2020/blob/cf006d133b9f3b9c6d89a2b2f21a71377c08f084/capstone/backend/src/main/java/com/google/servlets/GetItemTypesServlet.java#L53

Comment (chen-dawn) [reply capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java:127]: (and just ignore the page number param, since you're not using that).

Comment (luli0521) [reply capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java]: You can change the page param to be Optional: https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

File: capstone/backend/src/test/java/com/google/spanner/LibraryFunctionsTest.java [hunk 3]
```diff
@@ -119,9 +127,39 @@ public Void run(TransactionContext transaction) throws Exception {
...
+            "COOKIES",
+            "FLOUR",
+            "MILK",
+            "OLIVE OIL",
+            "PENCIL",
+            "RAMEN",
+            "SHAMPOO",
+            "SODA",
+            "SUGAR",
+            "WATER");
```
Comment (luli0521) [inline capstone/backend/src/test/java/com/google/spanner/LibraryFunctionsTest.java]: Can you extract the common items from this test and the test above to a constant list? And append "WATER" to that list in this test. It makes the code more readable.

Comment (officialcwang) [reply capstone/backend/src/test/java/com/google/spanner/LibraryFunctionsTest.java]: I ended up handling getting all items in the servlet instead (and kept the page functionality of the LibraryFunctions function), so I deleted this test.

File: capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java (hunk 1 above)
Comment (officialcwang) [reply capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java]: Ended up handling it in the servlet! I might come back and add this if I have time later, but I don't think it's necessary right now if I changed the API to its original function

**Pull request merged**